library-config/src/lib.rs

```diff
@@ -363,28 +396,75 @@ impl Configurator {
 
     pub fn get_config_from_file(
         &self,
-        path: &Path,
+        path_local: &Path,
+        path_managed: &Path,
         process_info: ProcessInfo<'_, impl Deref<Target = [u8]>>,
     ) -> anyhow::Result<Vec<LibraryConfig>> {
-        let stable_config = match fs::File::open(path) {
+        let stable_config_local = match fs::File::open(path_local) {
+            Ok(file) => self.parse_stable_config(&mut io::BufReader::new(file))?,
+            Err(e) if e.kind() == io::ErrorKind::NotFound => StableConfig::default(),
+            Err(e) => return Err(e).context("failed to open config file"),
+        };
+        let stable_config_managed = match fs::File::open(path_managed) {
             Ok(file) => self.parse_stable_config(&mut io::BufReader::new(file))?,
             Err(e) if e.kind() == io::ErrorKind::NotFound => StableConfig::default(),
             Err(e) => return Err(e).context("failed to open config file"),
         };
-        self.get_config(&stable_config, process_info)
+
+        let local_config = self.get_config(
+            &stable_config_local,
+            LibraryConfigSource::LocalFile,
+            &process_info,
+        );
+        let managed_config = self.get_config(
+            &stable_config_managed,
+            LibraryConfigSource::Managed,
+            &process_info,
+        );
+
+        let mut combined_config = local_config?;
+        combined_config.extend(managed_config?);
+        Ok(combined_config)
     }
 
     pub fn get_config_from_bytes(
         &self,
-        s: &[u8],
+        s_local: &[u8],
+        s_managed: &[u8],
         process_info: ProcessInfo<'_, impl Deref<Target = [u8]>>,
     ) -> anyhow::Result<Vec<LibraryConfig>> {
-        let stable_config: StableConfig = self.parse_stable_config(&mut io::Cursor::new(s))?;
-        self.get_config(&stable_config, process_info)
+        let stable_config_local: StableConfig =
+            self.parse_stable_config(&mut io::Cursor::new(s_local))?;
+        let stable_config_managed: StableConfig =
+            self.parse_stable_config(&mut io::Cursor::new(s_managed))?;
+
+        let local_config = self.get_config(
+            &stable_config_local,
+            LibraryConfigSource::LocalFile,
+            &process_info,
+        );
+
+        let managed_config = self.get_config(
+            &stable_config_managed,
+            LibraryConfigSource::Managed,
+            &process_info,
+        );
```

Comment:
Do we want both configs to be applied? I thought we'd match managed config, and only use the local config if no rule was matching in the managed one

Also if we do it like that, we should probably do some dedup in order to not return the same config twice

Reply:
Ah, yes, fair my bad.